src/app/payments/get-protocol-fee.ts

```diff
@@ -0,0 +1,129 @@
+import { decodeInvoice } from "@domain/bitcoin/lightning"
+import { checkedToWalletId } from "@domain/wallets"
+import {
+  LnPaymentRequestNonZeroAmountRequiredError,
+  LnPaymentRequestZeroAmountRequiredError,
+  PriceRatio,
+} from "@domain/payments"
+import { LndService } from "@services/lnd"
+import { PaymentFlowStateRepository } from "@services/payment-flow"
+import { WalletsRepository } from "@services/mongoose"
+import { NewDealerPriceService } from "@services/dealer-price"
+
+import {
+  constructPaymentFlowBuilder,
+  newCheckIntraledgerLimits,
+  newCheckWithdrawalLimits,
+} from "./helpers"
+
+export const getLightningFeeEstimation = async ({
+  walletId,
+  paymentRequest,
+}: {
+  walletId: string
+  paymentRequest: EncodedPaymentRequest
+}): Promise<PaymentAmount<WalletCurrency> | ApplicationError> => {
+  const decodedInvoice = decodeInvoice(paymentRequest)
+  if (decodedInvoice instanceof Error) return decodedInvoice
+  if (decodedInvoice.paymentAmount === null) {
+    return new LnPaymentRequestNonZeroAmountRequiredError()
+  }
+
+  return estimateLightningFee({
+    uncheckedSenderWalletId: walletId,
+    invoice: decodedInvoice,
+  })
+}
+
+export const getNoAmountLightningFeeEstimation = async ({
+  walletId,
+  paymentRequest,
+  amount,
+}: {
+  walletId: string
+  paymentRequest: EncodedPaymentRequest
```

Comment:
`walletId` and `number` has not been validated yet but `paymentRequest` has?

Reply:
is the type correct?

Reply:
Good point, it looks like we're pushing the validation into `decodeInvoice` in both send-payment and fee-probe use-case methods in which case the object is typed wrongly.

I can make a note to clean this up across those use-cases in a follow-up PR